backend/dataall/base/loader.py

```diff
@@ -237,15 +238,16 @@ def _check_loading_correct(in_config: Set[str], modes: Set[ImportMode]):
                 "Declare the module in depends_on"
             )
 
-    # 4) Checks all references for modules (when ModuleInterfaces don't exist or not supported)
+    # 4) Takes all loaded packages filtering those that support the current import modes it checks that it is expected to be loaded as a checked_module
     checked_module_names = {module.name() for module in expected_load}
-    # Modules from the config that doesn't support the current mode weren't added in Step1, adding them here
     checked_module_names |= in_config
-    for module in sys.modules.keys():
-        if module.startswith(_MODULE_PREFIX) and module != __name__:  # skip loader
-            name = _get_module_name(module)
-            if name and name not in checked_module_names:
-                raise ImportError(f"The package {module} has been imported, but it doesn't contain ModuleInterface")
+    dataall_packages_names = [_get_module_name(package) for package in sys.modules.keys() if package.startswith(_MODULE_PREFIX) and package != __name__]
+    for module in _all_modules():
+        if module.name() in dataall_packages_names and module.is_supported(modes):
+            if module.name() and module.name() not in checked_module_names:
```

Comment:
The loader works correctly in this scenario: It's complaining that we are trying to import `database_base` which shouldn't be loaded. Why is it happening? Before new `CDK_CLI_EXTENSION` mode, we had imported `database_base` directly in `datasets` in `__init__.py`.  Since `dataset` is part of the config, the loader doesn't raise an error while importing init file.  `dataset_base` is not standalone module and is no longer support all modes as it used to be, thus `__init__.py` shouldn't be initialized for `database_base`. 
The way to solve it: move global `dataset_base` imports to corresponding `ModuleInterface`s contructors